src/treecons.ml

```diff
@@ -0,0 +1,114 @@
+(** Tree constructors. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  { c_mod : string list
+  (** Module name where the symbol of the constructor is defined. *)
+  ; c_sym : string
+  (** Symbol of the constructor. *)
+  ; c_ari : int
+  (** Arity of the considered constructor.  A same symbol representation may
+      generate several constructors with different arities. *) }
```

Comment:
Do you have examples where you get different arities for a same symbol?

Reply:
No, not yet indeed.  But I tried to develop keeping in mind that it might happen (although I have never tested the case)

Reply:
Take f (c 0 x) -> r and f (c 1 x y) -> s, where c:(n:N)Fn and F:N=>TYPE is defined by F0 -> N and F(s n) -> N=>Fn.

Reply:
done as a trees_arity.lp file